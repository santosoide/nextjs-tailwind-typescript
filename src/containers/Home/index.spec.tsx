import React from 'react'
import { render } from '@testing-library/react'

import Index from './index'

describe('Index Page', () => {
  it('renders correctly', () => {
    const { asFragment } = render(<Index />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders hello', () => {
    // Given
    const EXPECTED = `Hello from ${process.env.SITE_NAME}.`

    const { getByTestId } = render(<Index />)

    // When
    const ACTUAL = getByTestId('helloH1').textContent

    // Then
    expect(ACTUAL).toEqual(EXPECTED)
  })

  it('renders hello with tailwind classes', () => {
    // Given
    const EXPECTED = 'text-xl text-gray-900'
    const { getByTestId } = render(<Index />)

    // When
    const ACTUAL = getByTestId('helloH1').className

    // Then
    expect(ACTUAL).toContain(EXPECTED)
  })
})
