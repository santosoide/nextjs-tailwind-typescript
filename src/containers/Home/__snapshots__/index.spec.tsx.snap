// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index Page renders correctly 1`] = `
<DocumentFragment>
  <h1
    class="text-xl text-gray-900"
    data-testid="helloH1"
  >
    Hello from 
    <u>
      Starter Project
    </u>
    .
  </h1>
</DocumentFragment>
`;
